packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h

```diff
@@ -0,0 +1,207 @@
+
+namespace fp{
+
+	class stratifiedInNodeClassIndicesUnsupervised : public stratifiedInNodeClassIndices
+	{
+		private:
+			std::vector<std::vector<int> > inSamples;
+			std::vector<int> inSamps;
+			std::vector<int> outSamps;
+			std::vector<std::vector<int> > outSamples;
+			std::vector<int> binSamples;
+			int inSampleSize;
+			int outSampleSize;
+
+			//TODO: the following functions would benefit from Vitter's Sequential Random Sampling
+		public:
+			stratifiedInNodeClassIndicesUnsupervised(): inSamples(fpSingleton::getSingleton().returnNumClasses()), outSamples(fpSingleton::getSingleton().returnNumClasses()), inSampleSize(0), outSampleSize(0){}
+
+
+			stratifiedInNodeClassIndicesUnsupervised(const int &numObservationsInDataSet): inSamples(fpSingleton::getSingleton().returnNumClasses()), outSamples(fpSingleton::getSingleton().returnNumClasses()), inSampleSize(0), outSampleSize(0){
+
+				createInAndOutSets(numObservationsInDataSet);
+				for(auto inSampsClass : inSamples){
+					inSampleSize += inSampsClass.size();
+				}
+
+				for(auto outSamps1 : outSamples){
+					outSampleSize += outSamps1.size();
+				}
+			}
+
+
+			inline void createInAndOutSets(const int &numObs){
+				std::vector<int> potentialSamples(numObs);
+
+				std::random_device rd; // obtain a random number from hardware
+				std::mt19937 eng(rd()); // seed the generator
+
+				std::uniform_int_distribution<> distr(0, numObs-1);
+
+				for(int i=0; i < numObs; ++i){
+					potentialSamples[i] = i;
+				}
+
+				int numUnusedObs = numObs;
+				int randomObsID;
+				int tempMoveObs;
+				for(int n=0; n<numObs; n++){
+					randomObsID = distr(eng);
+					inSamples[fpSingleton::getSingleton().returnLabel(potentialSamples[randomObsID])].push_back(potentialSamples[randomObsID]);
+					inSamps.push_back(potentialSamples[randomObsID]);
+					if(randomObsID < numUnusedObs){
+						--numUnusedObs;
+						tempMoveObs = potentialSamples[numUnusedObs];
+						potentialSamples[numUnusedObs] = potentialSamples[randomObsID];
+						potentialSamples[randomObsID] = tempMoveObs;
+					}
+				}
+
+				for(int n=0; n<numUnusedObs; ++n){
+					outSamples[fpSingleton::getSingleton().returnLabel(potentialSamples[randomObsID])].push_back(potentialSamples[n]);
+				
+					outSamps.push_back(potentialSamples[n]);
+					}
+				
+			}
+
+
+			inline double returnImpurity(){
+				if(false){
+					unsigned int sumClassTotalsSquared = 0;
+					for(auto i : inSamples){
+						sumClassTotalsSquared+=i.size()*i.size();
+					}
+					return 1-double(sumClassTotalsSquared)/(inSampleSize*inSampleSize);
+				}else{
+					double impSum = 0;
+					double classPercent;
+					for(auto i : inSamples){
+						classPercent = double(i.size())/double(inSampleSize);
+
+						impSum += double(i.size())*(1.0-classPercent);
+					}
+					return impSum;
+				}
+			}
+
+			inline void printIndices(){
+				std::cout << "samples in bag\n";
+				std::cout<<inSamples.size()<<"\n";
+				for( int n = 0; n < inSamps.size(); ++n){
```

Comment:
I get a compile time warning about the type mismatch between n and inSamps.size()

int vs. long unsigned int

Reply:
fixed